### PR TITLE
win_product_facts: Rewrite using AnsibleModule

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
Simple rewrite using AnsibleModule.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
win_product_facts

##### ANSIBLE VERSION
v2.8

### CODE REVIEW

File: lib/ansible/modules/windows/win_product_facts.ps1 [hunk 1]
```diff
@@ -3,19 +3,16 @@
...
-#$params = Parse-Args -arguments $args -supports_check_mode $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$result = @{
-    changed = $false
-    ansible_facts = @{
-        ansible_os_product_id = (Get-CimInstance Win32_OperatingSystem).SerialNumber
-    }
-}
+$module.Result.changed = $false
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_product_facts.ps1:15]: Not needed, set by default.

File: lib/ansible/modules/windows/win_product_facts.ps1 [hunk 2]
```diff
@@ -3,19 +3,16 @@
 # Copyright: (c) 2017, Dag Wieers (dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
 
-$ErrorActionPreference = "Stop"
+$spec = @{
+    options = @{}
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_product_facts.ps1:9]: You technically don't have to do this, by default this is what options is set to. Feel free to keep it here I just thought it best to mention.

File: lib/ansible/modules/windows/win_product_facts.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_product_facts.ps1:15]: Fixed in #48573

File: lib/ansible/modules/windows/win_product_facts.ps1 (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_product_facts.ps1:9]: Fixed in #48573

**Pull request merged**